paddle/fluid/pybind/tensor_py.h

```diff
@@ -13,7 +13,13 @@ See the License for the specific language governing permissions and
 limitations under the License. */
 
 #pragma once
+
+#ifndef NPY_NO_DEPRECATED_API
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#endif
+
 #include <Python.h>
+#include <numpy/arrayobject.h>
```

Comment:
If the constructor of `NumpyAllocation` only accepts `py::array` as its unique parameter, please remove this header.